Semana - 13/F4 - System/src/mainTaskManager.ts

```diff
@@ -0,0 +1,99 @@
+import { Specialty } from './specialty';
+import { MobileMission } from './mobileMission';
+import { WebMission } from './webmission';
+import { Teacher } from './teacher';
+import { Student } from './student';
+import { JSONFileManager } from './JSONFileManager';
+
+class MainTaskManager {
+  private webJSON = new JSONFileManager('webClass.json');
+  private mobileJSON = new JSONFileManager('mobileClass.json');
+
+  executeWeb(newClass: object) {
+    this.webJSON.saveToJSON(newClass);
+    this.pushTeacherWeb();
+    this.pushStudentWeb();
+  }
+
+  pushTeacherWeb() {
+    const register = this.webJSON.getJSONContent();
+    register.listTeachers.push(teacher1);
+    this.webJSON.saveToJSON(register);
+  }
+
+  pushStudentWeb() {
+    const register = this.webJSON.getJSONContent();
+    register.listStudents.push(newStudentWeb);
+    this.webJSON.saveToJSON(register);
+  }
+
+  executeMobile(newClass: object) {
+    this.mobileJSON.saveToJSON(newClass);
+    this.pushTeacherMobile();
+    this.pushStudentMobile();
+  }
+
+  pushTeacherMobile() {
+    const register = this.mobileJSON.getJSONContent();
+    register.listTeachers.push(teacher2);
+    this.mobileJSON.saveToJSON(register);
+  }
+
+  pushStudentMobile() {
+    const register = this.mobileJSON.getJSONContent();
+    register.listStudents.push(newStudentMobile);
+    this.mobileJSON.saveToJSON(register);
+  }
+}
+
+const webClass: Student[] = [];
+const mobileClass: Student[] = [];
+const teachers: Teacher[] = [];
+
+const newWebClass: WebMission = new WebMission(
+  'Bruce',
+  '06/01/2020',
```

Comment:
Pelo jeito, você está considerando que o curso dura 6 meses, né? Então, daria para você inicializar a data de início; e, a de término, você utiliza o `moment()` e adiciona 6 meses